File: translation_output/magicoder/codenet/java/python-sanitized/p03365.py
import sys
import io
import math

class MyScan:
    def __init__(self, inp):
        self.inp = inp

    def nextInt(self):
        num = 0
        minus = False
        b = self.inp.read(1)
        while b != '-' and not b.isdigit():
            b = self.inp.read(1)
        if b == '-':
            minus = True
            b = self.inp.read(1)
        while b.isdigit():
            num = num * 10 + int(b)
            b = self.inp.read(1)
        return -num if minus else num

class Util:
    _m = 1000000007
    fact = [1] * 1000001
    inv = [1] * 1000001
    ifact = [1] * 1000001

    @staticmethod
    def l():
        for i in range(2, len(Util.fact)):
            Util.fact[i] = i * Util.fact[i - 1] % Util._m
            Util.inv[i] = (Util._m - (Util._m // i) * Util.inv[Util._m % i] % Util._m) % Util._m
        Util.ifact = [1] + [Util.inv[i] * Util.ifact[i - 1] % Util._m for i in range(2, len(Util.fact))]

    @staticmethod
    def c(a, b):
        if Util.fact[a] == 1:
            Util.l()
        return Util.ifact[b] * Util.ifact[b - a] % Util._m * Util.fact[a] % Util._m

    @staticmethod
    def fact(a):
        if Util.fact[a] == 1:
            Util.l()
        return Util.fact[a]

class TaskC:
    def solve(self, testNumber, inp, out):
        i = inp.nextInt()
        md = [0] * i
        rm = i - 2
        lx = i - 2
        while rm >= 0:
            md[lx + 1] = Util.c(rm, lx)
            rm -= 2
            lx -= 1
        res = 0
        prev = 0
        for m in range(1, i):
            cur = Util.fact(m) * Util.fact(i - m - 1) % Util._m
            cur = cur * (md[m]) % Util._m
            res = (res + (cur - prev + Util._m) * m % Util._m) % Util._m
            prev = cur
        out.write(str(res) + '\n')

if __name__ == '__main__':
    inp = io.BytesIO(sys.stdin.read().encode())
    out = sys.stdout
    solver = TaskC()
    solver.solve(1, MyScan(inp), out)
    out.close()

File: translation_output/no_test/codellama/codenet/python/java-sanitized/p03365.java
import java.util.Scanner;

public class p03365 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] fact = new int[n + 1];
        int[] ifact = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            fact[i] = (fact[i - 1] * i) % MOD;
            ifact[i] = (int) Math.pow(fact[i], MOD - 2);
        }

        int ans = 0;
        int c0 = 0;
        for (int i = n / 2; i < n; i++) {
            int c1 = (nCk(i - 1, n - 1 - i) * fact[i] * fact[n - i - 1] - c0 + MOD) % MOD;
            ans = (ans + c1 * i) % MOD;
            c0 = (c0 + c1) % MOD;
        }

        System.out.println(ans);
    }

    public static int nCk(int n, int k) {
        if (k < 0 || k > n) {
            return 0;
        }
        return (fact[n] * ifact[n - k] * ifact[k]) % MOD;
    }
}